src/inc/til/ticket_lock.h

```diff
@@ -53,4 +53,89 @@ namespace til
         std::atomic<uint32_t> _next_ticket{ 0 };
         std::atomic<uint32_t> _now_serving{ 0 };
     };
+
+    struct recursive_ticket_lock
+    {
+        struct recursive_ticket_lock_suspension
+        {
+            constexpr recursive_ticket_lock_suspension(recursive_ticket_lock& lock) noexcept :
+                _lock{ lock }
+            {
+            }
+
+            ~recursive_ticket_lock_suspension()
+            {
+                if (_owner)
+                {
+                    _lock._lock.lock(); // lock-lock-lock lol
+                    _lock._owner.store(_owner, std::memory_order_relaxed);
```

Comment:
should the suspension be movable between threads?

should this type be copyable? movable?

Reply:
I don't totally understand why the suspension holds on to `_owner` -- if you suspend the lock and resume it on a different thread (unfortunately, due to coroutines, this might actually happen... unless you specifically disallow it!) `_owner` will be overwritten with the _original_ owner thread

Reply:
ah it's not copyable due to the lvalue reference member

Reply:
Yeah I explicitly don't want that (holding unchecked references across suspensions). I'll mark this class as non-copyable/moveable.